arelle/ValidateXbrlCalcs.py

```diff
@@ -49,61 +88,89 @@ def __init__(self, modelXbrl, inferDecimals=False, deDuplicate=False):
         self.conceptsInRequiresElement = set()
         
     def validate(self):
-        if not self.modelXbrl.contexts and not self.modelXbrl.facts:
-            return # skip if no contexts or facts
-        
-        if not self.inferDecimals: # infering precision is now contrary to XBRL REC section 5.2.5.2
-            self.modelXbrl.info("xbrl.5.2.5.2:inferringPrecision","Validating calculations inferring precision.")
+        modelXbrl = self.modelXbrl
+        xbrl21 = self.xbrl21
+        calc11 = self.calc11 # round or truncate
+        calc11t = self.calc11t # truncate
+        sumConceptItemRels = defaultdict(dict) # for calc11 dup sum-item detection
+          
+        if xbrl21:
+            if not self.modelXbrl.contexts and not self.modelXbrl.facts:
+                return # skip if no contexts or facts (note that in calc11 mode the dup relationships test is nonetheless required)
+            if not self.inferDecimals: # infering precision is now contrary to XBRL REC section 5.2.5.2
+                modelXbrl.info("xbrl.5.2.5.2:inferringPrecision","Validating calculations inferring precision.")
+        if calc11:
```

Comment:
Is there a reason not to use `elif` here (or below)?

Should there be a final `else: return` to handle `NONE`?

Reply:
elif and in suggestions done

A final return to handle NONE would defeat checking for essence-alias or  requires-element, both of which are independent of summation calc mode.